### ISSUE TITLE
Issue with csv files in windows system while they are under editing

### ISSUE THREAD

**Issue by rahulguptajss** [iterative/dvc#670]
I receive below error while csv is open and its modified. and dvc repro is executed.

$ dvc repro dvc/Dvcfile
Failed to reproduce 'dvc/Dvcfile': Failed to reproduce 'dvc\train.dvc': [WinError 6] The handle is invalid.

**Comment by efiop** [comment]
Could you run it again with `-v` flag and post logs here please?

**Comment by rahulguptajss** [comment]
$ dvc repro dvc/matrix.dvc -v
Data file or dir 'data\TestSet.csv' with cache '.dvc\cache\3e\3de0f9b0a1067bae03c182c845bf59' didn't change
Dvc file 'dvc\test.dvc' didn't change
Traceback (most recent call last):
  File "c:\programdata\anaconda3\lib\site-packages\dvc\project.py", line 149, in _reproduce_stages
    result += self._reproduce_stage(stages, n, force)
  File "c:\programdata\anaconda3\lib\site-packages\dvc\project.py", line 128, in _reproduce_stage
    stage = stages[node].reproduce(force=force)
  File "c:\programdata\anaconda3\lib\site-packages\dvc\stage.py", line 99, in reproduce
    if not self.changed() and not force:
  File "c:\programdata\anaconda3\lib\site-packages\dvc\stage.py", line 78, in changed
    if entry.changed():
  File "c:\programdata\anaconda3\lib\site-packages\dvc\output.py", line 232, in changed
    ret = self._changed_file(self.path, self.cache)
  File "c:\programdata\anaconda3\lib\site-packages\dvc\output.py", line 198, in _changed_file
    System.samefile(path, cache) and \
  File "c:\programdata\anaconda3\lib\site-packages\dvc\system.py", line 44, in samefile
    return winlink.samefile(path1, path2)
  File "c:\programdata\anaconda3\lib\site-packages\ntfsutils\hardlink.py", line 30, in samefile
    info1 = fs.getfileinfo(path1)
  File "c:\programdata\anaconda3\lib\site-packages\ntfsutils\fs.py", line 111, in getfileinfo
    return _getfileinfo(path, 0)
  File "c:\programdata\anaconda3\lib\site-packages\ntfsutils\fs.py", line 96, in _getfileinfo
    raise WinError()
OSError: [WinError 6] The handle is invalid.

Failed to reproduce 'dvc/matrix.dvc': Failed to reproduce 'dvc\train.dvc': [WinError 6] The handle is invalid.

**Comment by efiop** [comment]
Hm, so you are running `dvc repro` while you have a file dependency opened in another tab? If so, this is normal for Windows, since when you are modifying the file, its handle is changing, not allowing dvc to obtain file info using an older handle.

**Comment by efiop** [comment]
Though, this scenario is indeed not rare at all. We can probably try again if we didn't succeed to obtain file info the first time. I'll take a look shortly. Great catch, thank you!

**Comment by efiop** [comment]
Hm, interesting. Well, at least notepad is not affected. I'll try to reproduce it myself later and see if I can do something about it.

**Comment by efiop** [comment]
Ok, I was able to reproduce even without Excel. Looking into it right now.

**Comment by efiop** [comment]
Sorry that it took so long, only managed to get to it right now. After a bit more digging I was able to reproduce the bug and figured that it was caused by the exclusive access on the file and getfileinfo() from ntfsutils using GENERIC_READ flag for CreateFile(). I've sent a PR to that project https://github.com/sunshowers/ntfs/pull/11 with a fix. I'll introduce a temporary workaround for us soon while that PR is being considered by the maintainer.
